futures/tests/stream_futures_unordered.rs

```diff
@@ -406,3 +406,26 @@ fn clear_in_loop() {
         }
     });
 }
+
+// https://github.com/rust-lang/futures-rs/issues/2863#issuecomment-2219441515
+#[test]
+#[should_panic]
+fn panic_on_drop_fut() {
```

Comment:
Minimal example taken from issue. This failed under miri before the changes to fix it.